sql/core/src/test/scala/org/apache/spark/sql/VariantFunctionSuite.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.SparkThrowable
+import org.apache.spark.sql.catalyst.expressions.{CreateArray, CreateNamedStruct, Literal, StructsToJson}
+import org.apache.spark.sql.catalyst.expressions.variant.ParseJson
+import org.apache.spark.sql.execution.WholeStageCodegenExec
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.types.variant.VariantBuilder
+import org.apache.spark.types.variant.VariantUtil._
+import org.apache.spark.unsafe.types.VariantVal
+
+class VariantFunctionSuite extends QueryTest with SharedSparkSession {
+  import testImplicits._
+
+  test("parse_json") {
+    def check(json: String, expectedValue: Array[Byte], expectedMetadata: Array[Byte]): Unit = {
+      val df = Seq(json).toDF("v")
+      val variantDF = df.select(Column(ParseJson(Column("v").expr)))
+      val expected = new VariantVal(expectedValue, expectedMetadata)
+      checkAnswer(variantDF, Seq(Row(expected)))
+    }
+
```

Comment:
The test data is completely one-to-one corresponding to the original data in `VariantExpressionSuite`.

Reply:
They are slightly different. The previous tests are unit tests, using `checkEvaluation`. The new tests here are end-to-end tests, which are slower to run.

I'd suggest we test the implementation java function directly.

Reply:
Yeah, okay.

Reply:
I put the `type coercion` and `negative` of parseJson into the UT `VariantExpressionEvalUtilsSuite`.
The case involving `StructsToJson` be placed in UT `VariantFunctionSuite`, as it may be more suitable for testing in end-to-end scenarios.